File: 20220914-bestAndWorstLottos/app/src/main/java/BestAndWorstLottos.java
public class BestAndWorstLottos {
  public int[] solution(int[] lottos, int[] win_nums) {

    int[] answer = {};
    int getNumberOfWinNumbers = getRightNumber(lottos,win_nums);
    int numberOfZeros = countZero(lottos);
    int max = getNumberOfWinNumbers + numberOfZeros;
    int min = getNumberOfWinNumbers;

    answer = getRanks(min,max);
    return answer;
  }

  public int getRightNumber(int[] lottos, int[] win_nums) {
    int count = 0;
    for (int i = 0; i < lottos.length; i += 1) {
      for (int j = 0; j < win_nums.length; j += 1) {
        if (lottos[i] == win_nums[j]) {
          lottos[i] = -100;
          win_nums[j] = -1;
          count += 1;
        }
      }
    }
    return count;
  }

  public int countZero(int[] lottos) {
    int zeroCount = 0;
    for(int i : lottos) {
      if(i == 0) {
        zeroCount += 1;
      }
    }
    return zeroCount;
  }

  public int[] getRanks(int min, int max) {
    int[] answer;
    int minRank = 0;
    int rank = 0;
    for(int i = 6; i >= 1; i -= 1) {
      rank += 1;
      if(min == i) {
        minRank = rank;
      }
      if(min == 0) {
        minRank = 6;
      }
    }

    rank = 0;
    int maxRank = 0;
    for(int i = 6; i >= 1; i -= 1) {
      rank += 1;
      if(max == i) {
        maxRank = rank;
      }
      if(max == 0) {
        maxRank = 6;
      }
    }

    answer = new int[] {maxRank, minRank};
    return answer;
  }
}

File: 20220915-BestAndWorstLotto-JavaScript/BestAndWorstLottos.js
// 내 것과 당첨 표  일치번호 갯수구하기
// 0번호 갯수에 따른 최대 최소 구기
export function solution(lottos, winNums) {
  const rightNumbers = getRightNumbers(lottos, winNums);
  const zeroNumbers = getZeroNumbers(lottos);

  const maxRank = getMaxRank(rightNumbers, zeroNumbers);
  const minRank = getMinRank(rightNumbers);
  const answer = [maxRank, minRank];
  return answer;
}

export function getRightNumbers(lottos, winNums) {
  const rightNumbers = lottos.filter((lottosNumber) => winNums.includes(lottosNumber));
  return rightNumbers.length;
}

export function getZeroNumbers(lottos) {
  const zeroNumbers = lottos.filter((lottosNumber) => lottosNumber === 0);
  return zeroNumbers.length;
}

export function getMaxRank(rightNumbers, zeroNumbers) {
  const totalRights = rightNumbers + zeroNumbers;

  let rank = 0;
  let maxRank = 0;
  for (let i = 6; i >= 1; i -= 1) {
    rank += 1;
    if (totalRights === i) {
      maxRank = rank;
    }
    if (totalRights === 0) {
      maxRank = 6;
    }
  }

  return maxRank;
}

export function getMinRank(rightNumbers) {
  const totalRights = rightNumbers;

  let rank = 0;
  let minRank = 0;
  for (let i = 6; i >= 1; i -= 1) {
    rank += 1;
    if (totalRights === i) {
      minRank = rank;
    }
    if (totalRights === 0) {
      minRank = 6;
    }
  }

  return minRank;
}
